sdcm/cluster.py

```diff
@@ -3195,6 +3197,15 @@ def __init__(self, cluster_uuid=None, cluster_prefix='cluster', node_prefix='nod
             self.run_node_benchmarks()
         self.coredumps = {}
         super().__init__()
+        self.keyspace_info = {}
+
+    @lru_cache(maxsize=None)
+    def get_keyspace_info(self, keyspace_name: str, db_node: BaseNode):
+        if self.keyspace_info.get(keyspace_name, None) is None:
+            replication_strategy = ReplicationStrategy.get(db_node, keyspace_name)
+            self.keyspace_info[keyspace_name] = replication_strategy
+            logging.debug("Replication: %s", self.keyspace_info[keyspace_name])
+        return self.keyspace_info[keyspace_name]
```

Comment:
no need to save anything on the cluster anymore:
```suggestion
        replication_strategy = ReplicationStrategy.get(db_node, keyspace_name)
        return replication_strategy
```

Reply:
changed